compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3592,7 +3592,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     for (expected, actual) in zipped {
                         self.probe(|_| {
                             match self.at(&ObligationCause::misc(expr.span, body_id), param_env).eq(
-                                DefineOpaqueTypes::No,
+                                DefineOpaqueTypes::Yes,
```

Comment:
This is diagnostics code. Please leave a comment saying that and that `Yes` doesn't matter.